src/hooks/use_listen_message.rs

```diff
@@ -210,39 +210,41 @@ pub fn use_listen_message(cx: &ScopeState) -> &UseListenMessagesState {
                         });
                     }
                     info!("after write");
-                    // let room_name = if let Some(name) = message_event.room.name() {
-                    //     name
-                    // } else {
-                    //     let mut name = String::from("Unknown name room");
-                    //     let me = client.whoami().await.unwrap();
-                    //     let users = message_event.room.members().await;
-
-                    //     if let Ok(members) = users {
-                    //         let member = members
-                    //             .into_iter()
-                    //             .find(|member| !member.user_id().eq(&me.user_id));
-
-                    //         if let Some(m) = member {
-                    //             let n = m.name();
-
-                    //             name = String::from(n);
-                    //         }
-                    //     }
-
-                    //     name
-                    // };
-
-                    // handle_notification(
-                    //     NotificationItem {
-                    //         title: String::from(room_name),
-                    //         body: String::from(plain_message),
-                    //         show: true,
-                    //         handle: NotificationHandle {
-                    //             value: NotificationType::Click,
-                    //         },
-                    //     },
-                    //     notification.to_owned(),
-                    // );
+
+                    let room_name = match message_event.room.name() {
+                        Some(name) => name,
+                        None => {
+                            let mut name = String::from("Unknown name room");
+                            let me = client.whoami().await.unwrap();
```

Comment:
What happens on error here?

Reply:
Matching results is tedious and verbose, that's why the `?` operator was created, but to use it the block needs to return a result and the errors in the block need to be convertible to the result's error type. One trick to benefit from that operator in functions that can't return a result is to "pretend we return a result" in an inner block and handle the errors at the end, something like ...
```rust
async {
    // we can use the `?` operator as this block returns a result
    let foo = Err::<(), &'static str>("foo")?;
    // this line doesn't execute and there's no success as this always fails
    Ok(foo)
}
.unwrap_or_else(|s: &str| { // NOTE: this is from the future util crate
    // do something on error before we "ignore" it 
    info!("{s}");
});
```